From: Christoph Reiter <creiter@src.gnome.org>
Date: Mon, 19 Mar 2018 18:21:25 +0100
Subject: gtkstylecontext: guard against gtk_css_widget_node_get_widget()
 returning NULL in more places

When the widget gets finalized it clears the widgetnode and gtk_css_widget_node_get_widget
returns NULL. Guard against gtk_css_widget_node_get_widget() returning NULL like in other
places.

See https://gitlab.gnome.org/GNOME/pygobject/issues/28#note_82862

Origin: upstream, 3.22.30, commit:d65a4c2e0abeefb1e61e32a2da2c1e1ba9987817
---
 gtk/gtkstylecontext.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/gtk/gtkstylecontext.c b/gtk/gtkstylecontext.c
index 4ef6d57..2606815 100644
--- a/gtk/gtkstylecontext.c
+++ b/gtk/gtkstylecontext.c
@@ -187,7 +187,11 @@ gtk_style_context_real_changed (GtkStyleContext *context)
   GtkStyleContextPrivate *priv = context->priv;
 
   if (GTK_IS_CSS_WIDGET_NODE (priv->cssnode))
-    _gtk_widget_style_context_invalidated (gtk_css_widget_node_get_widget (GTK_CSS_WIDGET_NODE (priv->cssnode)));
+    {
+      GtkWidget *widget = gtk_css_widget_node_get_widget (GTK_CSS_WIDGET_NODE (priv->cssnode));
+      if (widget != NULL)
+        _gtk_widget_style_context_invalidated (widget);
+    }
 }
 
 static void
@@ -522,7 +526,8 @@ gtk_style_context_push_state (GtkStyleContext *context,
     {
       GtkWidget *widget = gtk_css_widget_node_get_widget (GTK_CSS_WIDGET_NODE (root));
       g_debug ("State %u for %s %p doesn't match state %u set via gtk_style_context_set_state ()",
-               state, gtk_widget_get_name (widget), widget, gtk_css_node_get_state (priv->cssnode));
+               state, (widget == NULL) ? "(null)" : gtk_widget_get_name (widget),
+               widget, gtk_css_node_get_state (priv->cssnode));
     }
   else
     {
